app/components/Admin/Room/RoomDetail.jsx

```diff
@@ -1,20 +1,90 @@
 "use strict"
+import R from 'ramda';
 import dom from '../../../utils/dom';
 import Room from './room-class.js'
+import Field from '../../Form/Field-class';
 
 var tempRoom = new Room();
 
-
-function saveRoom() {
+const {is, isEmpty, propEq, prop, all, filter, map, join, compose} = R;
+
+// Utils
+const allValid = all(prop('isValid'))
+const takeInvalidFields = filter(propEq('isValid', false));
+const getErrorMessages = map(prop('errorMsg'));
+const makeList = join('\r\n');
+const errorMessageList = compose(makeList, getErrorMessages, takeInvalidFields);
+
+// Validation
+const isNonEmptyString = (value) => is(String, value) && !isEmpty(value)
+const isNotNegative = (value) => +value >= 0;
+const isPositive = (value) => +value > 0;
+
+// Fields
+const RoomNumberInput = new Field({
+    name: 'number',
+    type: 'number',
+    errorMsg: 'Room number must be greater than zero!',
+    willDispatch: true,
+    validation: isPositive,
+    debug: true
+});
+
+const RoomNameInput = new Field({
+    name: 'name',
+    type: 'text',
+    errorMsg: 'Please enter a value for Room!',
+    willDispatch: true,
+    validation: isNonEmptyString,
+    debug: true
+});
+
+const RoomDescriptionInput = new Field({
+    name: 'description',
+    type: 'text',
+    willDispatch: true
+});
+
+const MinInput = new Field({
+    name: 'min',
+    type: 'number',
+    errorMsg: 'Minimum Capacity cannot be negative!',
+    willDispatch: true,
+    validation: isNotNegative
+});
+
+const MaxInput = new Field({
+    name: 'max',
+    type: 'number',
+    errorMsg: 'Maximum Capacity cannot be negative!',
+    willDispatch: true,
+    validation: isNotNegative
+});
+
+const ActiveInput = new Field({
```

Comment:
Added ActiveInput and adjusted the `FieldInput` class so it could handle type=checkbox